cibuildwheel/__main__.py

```diff
@@ -67,6 +68,19 @@ def main() -> None:
                             run in Windows, and it will build and test for all versions of
                             Python. Default: auto.
                         ''')
+
+    parser.add_argument(
+        '--archs',
+        default=None,
+        help='''
+            Comma-separated list of CPU architectures to build for.
+            If unspecified, builds the architectures natively supported
+            on this machine. Set this option to build an architecture
+            via emulation, for example, using binfmt_misc and qemu.
+            Default: auto
+            Choices: auto, {}
+        '''.format(", ".join(a.name for a in Architecture)),
```

Comment:
As expected, an enum.

That's horrible syntax IMO, looping over a type, but apparently [how Python envisions you doing things](https://docs.python.org/3/library/enum.html), so why not, then.

Reply:
This should be `a.value` rather than `a.name`, though, to match the logic from `Architecture.parse_config()` (even though, yes, names and values are the same) ?

Reply:
That's not a "type", it's an enum. How else would you loop over all the possible enums? C++ doesn't have this capability (yet, at least, until reflection is available?). Keep in mind adding any methods to Enum would remove possible Enum values! And yes, I think it should be `name.value`.

Reply:
Well, an enum is a type, no? Both in the broad sense (we're using it to tell mypy what's passed) as well as in the Python sense where `enum.EnumMeta` is a subclass of `type`.
But yeah, I can see it's a special one. Just haven't seen this that often and the capital/syntax highlighting is quite confusing; nothing this PR should do anything about, though!

Reiterating that `a.name` -> `a.value` is still a change to be made, just to make sure the serious part of my comments stands out ;-)

Reply:
Looking at the code, `CIProvider` really should not be a StrEnum - there's no need for that. I'm rather thinking `Architecture` shouldn't be either. It is _not_ needed to subclass `str` to _use_ strings as values, but only adds implicit conversion. I don't see any reason that we can't use explicit conversion here.

Though the implicit form does allow this beauty:

```python
", ".join(Architecture),
```

How's that for a "type"? :)

BTW, an enum should be thought of as both a type and a container of all the instances of that type. The fact that is is a type is actually a bit of a workaround in leu of true dedicated language support, for example `Architecture.count` is the `str.count` method; you can't add a "count" member to a StrEnum (at least without breaking its stringiness).

@joerick what do you think? Should I remove `str, ` from both of these and make them regular enums with string-based values, or do you want me to add:
```python
# Identical to StrEnum in Python 3.10 except for less checking on construction
# that we really are providing strings, and no support for enum.auto().
class StrEnum(str, Enum):
    __str__ = str.__str__
```
and then use that, making these full StrEnums?

Reply:
> How's that for a "type"? :)

Hah, still horrible, but better than before? ;-)

Reply:
I'm thinking the code might actually be clearer if we used plain old enums, rather than the StrEnum thing. Now we have type-checking, we end up having to be strict with them anyway. And `enum.value` is clearer and more explicit.